test/pytest/test_pcp.py

```diff
@@ -0,0 +1,671 @@
+                               reply_keys={'message': 'no "source" option specified for metrics channel'})
+    await transport.check_open('metrics1', source="bazinga", problem='not-supported',
+                               reply_keys={'message': 'unsupported "source" option specified for metrics: bazinga'})
+    await transport.check_open('metrics1', source="/non-existant", problem='not-found')
+    await transport.check_open('metrics1', source=str(archive),
+                               metrics=[{"name": ""}],
+                               problem='protocol-error',
+                               reply_keys={'message': 'invalid "metrics" option was specified (no name for metric)'})
+    await transport.check_open('metrics1', source=str(archive), limit=-10, problem='protocol-error',
+                               reply_keys={'message': 'invalid "limit" value: -10'})
+    invalid_timestamp_size = sys.maxsize * 1000 + 3
+    await transport.check_open('metrics1', source=str(archive), timestamp=invalid_timestamp_size,
+                               problem='protocol-error',
+                               reply_keys={'message': f'invalid "timestamp" value: {invalid_timestamp_size}'})
+    await transport.check_open('metrics1', problem='protocol-error',
+                               reply_keys={'message': 'no "source" option specified for metrics channel'})
+
+
+@pytest.mark.asyncio
+async def test_pcp_open(transport, archive):
+    _ = await transport.check_open('metrics1', source=str(archive),
+                                    metrics=[{"name": "mock.value"}])
+
+    _, data = await transport.next_frame()
+    # first message is always the meta message
+    meta = json.loads(data)
+    # C bridge
+    # {"timestamp":0,"now":1708092219642,"interval":1000,
+    #  "metrics":[{"name":"mock.value","units":"","semantics":"instant"}]}
+
+    assert_metrics_meta(meta, str(archive))
+
+    metrics = meta['metrics']
+    assert len(metrics) == 1
+
+    metric = metrics[0]
+    assert metric['name'] == 'mock.value'
+    assert 'derive' not in metric
+    assert metric['semantics'] == 'instant'
+
+    # assert_sample (tc, "[[10],[11],[12]]");
+    _, data = await transport.next_frame()
+    data = json.loads(data)
+    assert data == [[10], [11], [12]]
+
+    _, data = await transport.next_frame()
+    # first message is always the meta message
+    meta = json.loads(data)
+
+    metrics = meta['metrics']
+    assert len(metrics) == 1
+
+    metric = metrics[0]
+    assert metric['name'] == 'mock.value'
+    assert 'derive' not in metric
+    assert metric['semantics'] == 'instant'
+
+    # C bridge sends a META message per archive
+
+    # assert_sample (tc, "[[13],[14],[15]]");
+    _, data = await transport.next_frame()
+    data = json.loads(data)
+    assert data == [[13], [14], [15]]
+
+
+@pytest.mark.asyncio
+async def test_pcp_big_archive(transport, big_archive):
+    _ = await transport.check_open('metrics1', source=str(big_archive),
```

Comment:
Why the weird `_ =`?

Reply:
Leftover from refactoring I suppose, dropped